packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -847,28 +854,27 @@ class DebuggerController extends DisposableController
   ///
   /// The return value can be one of [Obj] or [Sentinel].
   Future<Obj> getObject(ObjRef objRef) {
-    return _service.getObject(isolateRef.id, objRef.id);
+    return _service.getObject(isolateRef.id!, objRef.id!);
   }
 
   Future<void> _populateScripts(Isolate isolate) async {
-    assert(isolate != null);
     final scriptRefs = await scriptManager.retrieveAndSortScripts(isolateRef);
 
     // Update the selected script.
-    final mainScriptRef = scriptRefs.firstWhere((ref) {
-      return ref.uri == isolate.rootLib.uri;
-    }, orElse: () => null);
+    final mainScriptRef = scriptRefs.firstWhereOrNull((ref) {
+      return ref.uri == isolate.rootLib!.uri;
+    })!;
```

Comment:
null asserting a `firstWhereOrNull` return is some null-safety code smell. Remove the null-assert here and either handle the null case here or make sure _populateScriptAndShowLocation handles a null value